src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.Windows.cs

```diff
@@ -18,12 +18,67 @@ public static IEnumerable<object[]> IsComImport_Windows_ReturnsExpected()
         }
 
         [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
-        [MemberData(nameof(IsComImport_Windows_ReturnsExpected))]
+        [MemberData(nameof(IsComObject_Windows_TestData))]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Not approved COM object for app")]
         public void IsComObject_Windows_ReturnsExpected(object value, bool expected)
         {
             Assert.Equal(expected, Marshal.IsComObject(value));
         }
+
+        [Fact]
+        public void GenerateGuidForType_ComObject_ReturnsComGuid()
+        {
+            Assert.Equal(new Guid("927971f5-0939-11d1-8be1-00c04fd8d503"), Marshal.GenerateGuidForType(typeof(ComImportObject)));
+        }
+
+        [Fact]
+        public void GenerateProgIdForType_ImportType_ThrowsArgumentException()
+        {
+            AssertExtensions.Throws<ArgumentException>("type", () => Marshal.GenerateProgIdForType(typeof(ComImportObject)));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "Marshal.GetComObjectData is not implemented in .NET Core.")]
+        public void GetComObjectData_NetFramework_ReturnsExpected()
+        {
+            Type type = Type.GetTypeFromCLSID(new Guid("927971f5-0939-11d1-8be1-00c04fd8d503"));
+            object comObject = Activator.CreateInstance(type);
+
+            Assert.Null(Marshal.GetComObjectData(comObject, "key"));
+
+            Marshal.SetComObjectData(comObject, "key", 1);
+            Assert.Equal(1, Marshal.GetComObjectData(comObject, "key"));
+            Assert.Null(Marshal.GetComObjectData(comObject, "noSuchKey"));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "Marshal.GetIDispatchForObject is not implemented in .NET Core.")]
+        public static void GetIDispatchForObject_ComObject_ReturnsNonZero()
+        {
+            Type type = Type.GetTypeFromCLSID(new Guid("927971f5-0939-11d1-8be1-00c04fd8d503"));
```

Comment:
Nit: Could you create a separate file for each API? such as GetIDispatchForObject.cs which contains Marshal.GetIDispatchForObject() testing scenario?

Reply:
Done